.github/workflows/build.yml

```diff
@@ -21,12 +21,30 @@ jobs:
         go-arch: ["amd64", "arm", "arm64"]
     steps:
       - uses: actions/checkout@v4
-      - uses: DeterminateSystems/nix-installer-action@main
-      - uses: DeterminateSystems/magic-nix-cache-action@main
       - uses: actions/setup-go@v5
         with:
           go-version: "1.23"
           check-latest: true
+      - name: Cache libraries
+        id: cache-rocksdb
+        if: env.GIT_DIFF
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-rocksdb-lib
+        with:
+          path: /usr/local/lib
+          key: ${{ runner.os }}-rocksdb-${{ matrix.go-arch }}
+      - name: install rocksdb
+        if: env.GIT_DIFF && steps.cache-rocksdb.outputs.cache-hit != 'true'
```

Comment:
As rocksdb compilation lasts ~8mn on public runners, it will be executed only when cache is not invalited by github (should be once every 7 days)